### DISCUSSION TITLE
Notifcations -- Container names that require updates

### DISCUSSION THREAD

**Discussion by liquidsunshine** [containrrr/watchtower#1614] [Q&A]
Hello!

I'm using Watchtower for update notifications, but not automatic updates.  I want to have a chance to read the changelogs before performing updates.

As such, it would be very helpful to know which containers require updates.

Right now I get emails like this:
```
Found new postgres:15-alpine image (ddc12ac7fa27)
Found new nginx:alpine image (8e75cbc5b25c)
```

But since I use MariaDB, Postgres, and nginx in several stacks (sometimes with different versions; sometimes with Alpine and sometimes a different base) I end up having to do some hackery to figure out which containers actually use those images.  

Is there a way to get output closer to this?:
```
Found new postgres:15-alpine image (appname_postgres: 15.1 -> 15.2)
Found new nginx:alpine image (appname_nginx: 1.20 -> 1.22)
```
(where appname_postgres and appname_nginx are specific containers that use the alpine flavors of those containers)

I didn't find any previous discussions about this, and it wasn't clear how to do this from the notifications documentation in Watchtower.  Apologies if I missed something obvious.

**Comment by piksel** [comment]
Almost. You can get a list of such containers with newer images using a combination of `--notification-report` and `--monitor-only`, and setting the notification template to only show the `stale` containers:

```sh
docker run -d \
	--name watchtower \
	-v /var/run/docker.sock:/var/run/docker.sock \
	containrrr/watchtower \
	--monitor-only \
	--notification-report \
	--notification-template '
{{- with .Report -}}
  {{- range .Stale -}}
    Found new {{.ImageName}} image ({{.Name}}: {{.CurrentImageID.ShortID}} -> {{.LatestImageID.ShortID}})
  {{- end -}}
{{- end -}}
'
```
Result:
```
Found new postgres:15.2 image (/testapp-db: 80c558ffdc31 -> ccd94e8b5fd9)
```

Watchtower has no knowledge of any versions of the software inside the containers though...
